pkg/genericapiserver/config.go

```diff
@@ -0,0 +1,467 @@
+	s.HandlerContainer.Router(restful.CurlyRouter{})
+	s.MuxHelper = &apiserver.MuxHelper{Mux: s.mux, RegisteredPaths: []string{}}
+
+	if c.ProxyDialer != nil || c.ProxyTLSClientConfig != nil {
+		s.ProxyTransport = utilnet.SetTransportDefaults(&http.Transport{
+			Dial:            c.ProxyDialer,
+			TLSClientConfig: c.ProxyTLSClientConfig,
+		})
+	}
+
+	// Register root handler.
+	// We do not register this using restful Webservice since we do not want to surface this in api docs.
+	// Allow GenericAPIServer to be embedded in contexts which already have something registered at the root
+	if c.EnableIndex {
+		s.mux.HandleFunc("/", apiserver.IndexHandler(s.HandlerContainer, s.MuxHelper))
+	}
+
+	if c.EnableLogsSupport {
+		apiserver.InstallLogsSupport(s.MuxHelper, s.HandlerContainer)
+	}
+	if c.EnableUISupport {
+		ui.InstallSupport(s.MuxHelper, s.enableSwaggerSupport && s.enableSwaggerUI)
+	}
+
+	if c.EnableProfiling {
+		s.mux.HandleFunc("/debug/pprof/", pprof.Index)
+		s.mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+		s.mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	}
+
+	apiserver.InstallVersionHandler(s.MuxHelper, s.HandlerContainer)
+
+	handler := http.Handler(s.mux.(*http.ServeMux))
+
+	// TODO: handle CORS and auth using go-restful
+	// See github.com/emicklei/go-restful/blob/master/examples/restful-CORS-filter.go, and
+	// github.com/emicklei/go-restful/blob/master/examples/restful-basic-authentication.go
+
+	if len(c.CorsAllowedOriginList) > 0 {
+		allowedOriginRegexps, err := util.CompileRegexps(c.CorsAllowedOriginList)
+		if err != nil {
+			glog.Fatalf("Invalid CORS allowed origin, --cors-allowed-origins flag was set to %v - %v", strings.Join(c.CorsAllowedOriginList, ","), err)
+		}
+		handler = apiserver.CORS(handler, allowedOriginRegexps, nil, nil, "true")
+	}
+
+	s.InsecureHandler = handler
+
+	attributeGetter := apiserver.NewRequestAttributeGetter(s.RequestContextMapper, s.NewRequestInfoResolver())
+	handler = apiserver.WithAuthorizationCheck(handler, attributeGetter, s.authorizer)
+	if len(c.AuditLogPath) != 0 {
+		// audit handler must comes before the impersonationFilter to read the original user
+		writer := &lumberjack.Logger{
+			Filename:   c.AuditLogPath,
+			MaxAge:     c.AuditLogMaxAge,
+			MaxBackups: c.AuditLogMaxBackups,
+			MaxSize:    c.AuditLogMaxSize,
+		}
+		handler = audit.WithAudit(handler, s.RequestContextMapper, writer)
+		defer writer.Close()
+	}
+	handler = apiserver.WithImpersonation(handler, s.RequestContextMapper, s.authorizer)
+
+	// Install Authenticator
+	if c.Authenticator != nil {
+		authenticatedHandler, err := handlers.NewRequestAuthenticator(s.RequestContextMapper, c.Authenticator, handlers.Unauthorized(c.SupportsBasicAuth), handler)
+		if err != nil {
+			glog.Fatalf("Could not initialize authenticator: %v", err)
+		}
+		handler = authenticatedHandler
+	}
+
+	// TODO: Make this optional?  Consumers of GenericAPIServer depend on this currently.
+	s.Handler = handler
```

Comment:
> Fixing, but I'll point out that I was shooting for a straight move.

hrm. side effects will be subtle if I break it doing this.  Can I get a deferment so you can carefully review 10 lines instead of a bulk move?